http-clients/aws-crt-client/src/main/java/software/amazon/awssdk/http/crt/internal/CrtRequestExecutor.java

```diff
@@ -106,11 +174,26 @@ private static void reportMetrics(CrtRequestContext executionContext, MetricColl
         metricCollector.reportMetric(PENDING_CONCURRENCY_ACQUIRES, saturatedCast(managerMetrics.getPendingConcurrencyAcquires()));
     }
 
-    private void executeRequest(CrtRequestContext executionContext,
+    private static void reportSyncMetrics(CrtRequestContext executionContext, MetricCollector metricCollector,
```

Comment:
This method seems to be the same as `reportAsyncMetrics` except one takes executionContext and the other takes CrtAsyncRequestContext. Can we pass `HttpClientConnectionManager` instead so that we can reuse this method in both places?

Reply:
Good call! I was looking for ways to simplify that and that's way simpler.